lib/puppet/provider/augeas/augeas.rb

```diff
@@ -570,4 +571,9 @@ def do_execute_changes
       end
     end
   end
+
+  def to_array(string)
+    JSON.parse(string.tr("'", '"'))
```

Comment:
This simple substitution means it's impossible to match against values that contain single quotes / apostrophes. Are we comfortable with losing that capability?

Reply:
That's a good question, I need to look at that more. Also need to make changes to the puppetlabs-augeas_core module too.